app/utils/scroll-to-error.ts

```diff
@@ -0,0 +1,32 @@
+export const scrollToError = () => {
+  const scrollToFirstError = () => {
+    const errorElements = document.querySelectorAll(".text-error-500");
+    if (errorElements.length > 0) {
+      let firstError = errorElements[0];
+      const elementToScrollTo = firstError.previousElementSibling || firstError;
+      elementToScrollTo.scrollIntoView({ behavior: "smooth" });
+    }
+  };
+
+  scrollToFirstError();
+
+  const observer = new IntersectionObserver(
```

Comment:
I am having trouble to understand how this works. I see the observer and I understand that part but how does it interact with `scrollToError`. Seems like scroll will always be called on all entries, not just one the ones that are not visible.

Also please update this code by adding comments how it works, it needs to be more clear to developers who have to use and interact with this.